app/src/main/java/org/oppia/app/player/state/InteractionAdapter.kt

```diff
@@ -0,0 +1,160 @@
+import org.oppia.app.databinding.MultipleChoiceInteractionItemsBinding
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.model.StringList
+import org.oppia.app.player.state.itemviewmodel.CustomizationArgsInteractionViewModel
+import org.oppia.app.player.state.itemviewmodel.SelectionContentViewModel
+import org.oppia.app.player.state.listener.InteractionAnswerRetriever
+import org.oppia.util.parser.HtmlParser
+
+private const val VIEW_TYPE_MULTIPLE_CHOICE = 1
+private const val VIEW_TYPE_ITEM_SELECTION = 2
+private const val INTERACTION_ADAPTER_TAG = "Interaction Adapter"
+
+/** Adapter to bind the interactions to the [RecyclerView]. It handles MultipleChoiceInput and ItemSelectionInput interaction views. */
+class InteractionAdapter(
+  private val htmlParserFactory: HtmlParser.Factory,
+  private val entityType: String,
+  private val explorationId: String,
+  private val itemList: MutableList<SelectionContentViewModel>,
+  private val customizationArgs: CustomizationArgsInteractionViewModel
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>(), InteractionAnswerRetriever {
+
+  private var itemSelectedPosition = -1
+  private var selectedAnswerIndex = -1
+  private var selectedHtmlStringList = mutableListOf<String>()
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+    return when (viewType) {
+      VIEW_TYPE_MULTIPLE_CHOICE -> {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding =
+          DataBindingUtil.inflate<MultipleChoiceInteractionItemsBinding>(
+            inflater,
+            R.layout.multiple_choice_interaction_items,
+            parent,
+            /* attachToParent= */ false
+          )
+        MultipleChoiceViewHolder(binding)
+      }
+      VIEW_TYPE_ITEM_SELECTION -> {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding =
+          DataBindingUtil.inflate<ItemSelectionInteractionItemsBinding>(
+            inflater,
+            R.layout.item_selection_interaction_items,
+            parent,
+            /* attachToParent= */ false
+          )
+        ItemSelectionViewHolder(binding)
+      }
+      else -> throw IllegalArgumentException("Invalid view type")
+    }
+  }
+
+  override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+    when (holder.itemViewType) {
+      VIEW_TYPE_MULTIPLE_CHOICE -> (holder as MultipleChoiceViewHolder).bind(
+        itemList[position].htmlContent,
+        position,
+        itemSelectedPosition
+      )
+      VIEW_TYPE_ITEM_SELECTION -> (holder as ItemSelectionViewHolder).bind(
+        itemList[position]
+      )
+    }
+  }
+
+  // Determines the appropriate ViewType according to the interaction type.
+  override fun getItemViewType(position: Int): Int {
+    return if (customizationArgs.interactionId == "ItemSelectionInput") {
+      if (customizationArgs.maxAllowableSelectionCount > 1) {
+        VIEW_TYPE_ITEM_SELECTION
```

Comment:
It's slightly confusing that we can use the multiple choice type for an item selection interaction. Suggest instead distinguishing between the input types, e.g. ``VIEW_TYPE_CHECKBOXES`` vs. ``VIEW_TYPE_RADIO_BUTTONS``

Reply:
Done.